lib/compat/wordpress-6.5/block-patterns.php

```diff
@@ -31,3 +31,50 @@ function gutenberg_register_media_pattern_categories() {
 	);
 }
 add_action( 'init', 'gutenberg_register_media_pattern_categories' );
+
+/**
+ * Adds a new taxonomy for organizing user created patterns.
+ *
+ * Note: This should be removed when the minimum required WP version is >= 6.4.
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/53163
+ *
+ * @return void
+ */
+function gutenberg_register_taxonomy_patterns() {
+	$args = array(
+		'public'             => false,
+		'publicly_queryable' => false,
+		'hierarchical'       => false,
+		'labels'             => array(
+			'name'                       => _x( 'Pattern Categories', 'taxonomy general name' ),
+			'singular_name'              => _x( 'Pattern Category', 'taxonomy singular name' ),
+			'add_new_item'               => __( 'Add New Pattern Category' ),
```

Comment:
What do you think about making this "Add new category" instead? It's a bit duplicative as-is: 

![CleanShot 2023-12-15 at 17 14 36](https://github.com/WordPress/gutenberg/assets/1813435/6bf99578-f4e0-4ba1-aba3-f458d9953b1b)